systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java

```diff
@@ -171,48 +175,20 @@ void testTlsExternalUserWithQuotas(ExtensionContext extensionContext) {
         // skip test if KRaft mode is enabled and Kafka version is lower than 3.5.0 - https://github.com/strimzi/strimzi-kafka-operator/issues/8806
         assumeFalse(Environment.isKRaftModeEnabled() && TestKafkaVersion.compareDottedVersions("3.5.0", Environment.ST_KAFKA_VERSION) == 1);
 
-        final String kafkaUserName = testStorage.getKafkaUsername();
-        final KafkaUser tlsExternalUser = KafkaUserTemplates.tlsExternalUser(Environment.TEST_SUITE_NAMESPACE, userClusterName, kafkaUserName).build();
+        final KafkaUser tlsExternalUser = KafkaUserTemplates.tlsExternalUser(Environment.TEST_SUITE_NAMESPACE, userClusterName, testStorage.getKafkaUsername()).build();
 
         testUserWithQuotas(extensionContext, tlsExternalUser);
     }
 
     @ParallelTest
     void testScramUserWithQuotas(ExtensionContext extensionContext) {
         KafkaUser user = KafkaUserTemplates.scramShaUser(Environment.TEST_SUITE_NAMESPACE, userClusterName, "scramed-arnost").build();
-
         testUserWithQuotas(extensionContext, user);
     }
 
-    @ParallelTest
-    void testUserTemplate(ExtensionContext extensionContext) {
+    synchronized void testUserWithQuotas(ExtensionContext extensionContext, KafkaUser user) {
```

Comment:
Why does this need to be synchronized? I know that's not the aim of this PR but I'm just curious. I glance at this code inside the method I do not see anything, which is not thread-safe. 

1. storageMap is ConcurrentHashMap so this easily handle concurrent access
2. resourceManager is thread-safe by his implementation

and other usages of objects like KafkaCmdClient, KafkaClients, KafkaUserResource, SecretUtils, ClientUtils, and TestUtils do not modify any shared state, which could cause some concurrent problems.

Did you try to execute these tests in parallel? Again such change should be done in another PR but I just wanted to point on it :)

Reply:
This is just diff where i  deleted test, i did not add the synchronized, but i can remove it though, tried it without it, seems to work. Thanks